---
title: "Cdogr"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Cdogr}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```
# Loading the package

Load `Cdogr` with:

```{r setup}
library(Cdogr)
```
# Introduction

`Cdogr` is a collection of functions designed to work with DNA and RNA sequences. It allows you to generate random DNA sequences, convert DNA to RNA, split RNA into codons, translate RNA codons to amino acids, and visualize the frequency of amino acids in a sequence.


# Functions in `Cdogr`

## 1. Generating a Random DNA Sequence

You can generate a random DNA sequence of a specified length with the `generate_dna_sequence` function.

```r
dna_sequence <- generate_dna_sequence(length = 30)
print(dna_sequence)
```

## 2. Converting DNA to RNA

You can convert a DNA sequence to RNA with the `dna_to_rna` function.

```r
rna_sequence <- dna_to_rna(dna_sequence)
print(rna_sequence)
```

## 3. Splitting RNA into Codons

You can split an RNA sequence into its constituent codons with the `split_into_codons` function.

```r
codons <- split_into_codons(rna_sequence, start = 1)
print(codons)
```

## 4. Translating RNA Codons to Amino Acids

You can translate a sequence of RNA codons into amino acids with the `translate_codons` function.

```r
amino_acids <- translate_codons(codons)
print(amino_acids)
```

## 5. Visualizing Amino Acid Frequencies

You can visualize the frequency of each amino acid in a sequence with the `plot_amino_acid_frequencies` function.

```r
plot <- plot_amino_acid_frequencies(amino_acids)
print(plot)
```

# Conclusion

`Cdogr` provides a suite of functions for working with DNA and RNA sequences, from generating random sequences to visualizing amino acid frequencies. We hope that this package will be a valuable tool in your bioinformatics analyses.
